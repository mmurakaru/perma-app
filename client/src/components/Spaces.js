import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { ReactComponent as Plus } from '../assets/plus.svg'
import { ReactComponent as Sunflower } from '../assets/sunflower.svg'

SpaceOverview.propTypes = {
    spaces: PropTypes.array,
    updateSpaceDetails: PropTypes.func,
    switchToSpaceCreator: PropTypes.func,
}

export default function SpaceOverview({
    spaces,
    updateSpaceDetails,
    switchToSpaceCreator,
}) {
    return (
        <>
            <SpacesContainer>
                {spaces.map(({ spaceTitle, id }) => (
                    <SpaceWrapper
                        key={id}
                        onClick={() => updateSpaceDetails(id)}
                    >
                        <SpaceIcon>
                            <LogoStyled />
                        </SpaceIcon>
                        <h2>{spaceTitle}</h2>
                    </SpaceWrapper>
                ))}

                <SpaceWrapper onClick={switchToSpaceCreator}>
                    <NewSpaceIcon>
                        <PlusIcon />
                    </NewSpaceIcon>
                    <h2>Create a new space</h2>
                </SpaceWrapper>
            </SpacesContainer>
        </>
    )
}

const SpacesContainer = styled.div`
    margin-top: 50px;
    height: 78.5vh;
    display: grid;
    gap: 15px;
    grid-template-columns: 1fr 1fr 1fr;
    grid-auto-rows: min-content;
    place-items: center;
    overflow: auto;

    h2 {
        margin-left: 20px;
        font-weight: 400;
        font-size: 0.6rem;
        opacity: 50%;
    }
`

const NewSpaceIcon = styled.div`
    border: 1px solid var(--medium-grey);
    height: 100px;
    width: 100px;
    border-radius: 5px;
    display: grid;
    place-items: center;
`

const PlusIcon = styled(Plus)`
    width: 50%;
    display: block;
    fill: var(--dark-grey);
`

const SpaceIcon = styled(NewSpaceIcon)`
    border: none;
    background-color: var(--light-green);
`
const LogoStyled = styled(Sunflower)`
    width: 80%;
`

const SpaceWrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-self: start;

    h2 {
        margin: 3px 0;
        opacity: 100%;
    }
`
