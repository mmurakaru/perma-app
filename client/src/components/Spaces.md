Spaces example:

```js
<Spaces spaces={['Vegetables', 'Fruits']} />
```
