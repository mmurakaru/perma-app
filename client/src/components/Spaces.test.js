import Spaces from './Spaces'
import { render } from '@testing-library/react'

describe('SpacesOverview', () => {
    it('renders correctly', () => {
        const { container } = render(
            <Spaces spaces={['Vegetables', 'Fruits']} />
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})
