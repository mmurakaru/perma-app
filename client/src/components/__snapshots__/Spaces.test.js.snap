// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpacesOverview renders correctly 1`] = `
.c0 {
  margin-top: 50px;
  height: 78.5vh;
  display: grid;
  gap: 15px;
  grid-template-columns: 1fr 1fr 1fr;
  grid-auto-rows: min-content;
  place-items: center;
  overflow: auto;
}

.c0 h2 {
  margin-left: 20px;
  font-weight: 400;
  font-size: 0.6rem;
  opacity: 50%;
}

.c2 {
  border: 1px solid var(--medium-grey);
  height: 100px;
  width: 100px;
  border-radius: 5px;
  display: grid;
  place-items: center;
}

.c5 {
  width: 50%;
  display: block;
  fill: var(--dark-grey);
}

.c3 {
  border: none;
  background-color: var(--light-green);
}

.c4 {
  width: 80%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  justify-self: start;
}

.c1 h2 {
  margin: 3px 0;
  opacity: 100%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    >
      <svg
        class="c4"
      >
        sunflower.svg
      </svg>
    </div>
    <h2>
      spaceTitle
    </h2>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    >
      <svg
        class="c4"
      >
        sunflower.svg
      </svg>
    </div>
    <h2>
      spaceTitle
    </h2>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        class="c5"
      >
        plus.svg
      </svg>
    </div>
    <h2>
      Create a new space
    </h2>
  </div>
</div>
`;
